src/export/digitalPlanning/index.ts

```diff
@@ -12,7 +13,9 @@ export async function generateDigitalPlanningPayload({
   client,
   sessionId,
   skipValidation,
-}: DigitalPlanningExportParams): Promise<DigitalPlanningPayload> {
+}: DigitalPlanningExportParams): Promise<
+  ApplicationPayload | PreApplicationPayload
+> {
```

Comment:
In an ideal world, I think we'd have two separate classes that each return a precise type rather than a single class that returns `ApplicationPayload | PreApplicationPayload` - but this approach seemed like the path of least resistance to share as many of the existing passport helper methods as possible without a massive refactor or duplication (eg they're `private` functions!)

Reply:
Agreed on general approach here for sure.

If we hit any type issues with this in planx-new in the short term, we could maybe consider passing an additional argument into the class - 

```ts
type DigitalPlanningExportParams = {
  client: Client;
  sessionId: string;
  skipValidation?: boolean;
  type: "preApp" | ApplictionTypeUnion; // Create type discriminator here
};

export async function generateDigitalPlanningPayload({
  client,
  sessionId,
  skipValidation,
  type,
}: DigitalPlanningExportParams): Promise
 //  Modify return type based on this
  type extends "preApp'"? PreApplicationPayload : ApplicationPayload
> {
  ...
}
```

Returning a union though is exactly what I would have done here - I don't think we use the return type much in PlanX at this stage.

Reply:
Appreciate you coming back to this one ! There's an explicit reason I haven't gone with an additional argument from the planx-new side which is the (admittedly awkward) reality that discretionary/undefined-application-type payloads are also getting routed through this function to create a generic "application.json" for submission payloads (send to email & S3).

Therefore the discriminated type exposed to planx-new wouldn't be so tidy, but have to be something like this which felt pretty meaningless! 
```
"preApp" | ApplicationTypeUnion | string | undefined
```